kpi/paginators.py

```diff
@@ -1,7 +1,84 @@
 # -*- coding: utf-8 -*-
-from rest_framework import serializers, exceptions
-from rest_framework.pagination import LimitOffsetPagination, PageNumberPagination
+from django.core.paginator import InvalidPage
+from rest_framework import serializers
+from rest_framework.exceptions import NotFound
+from rest_framework.pagination import (
+    LimitOffsetPagination,
+    PageNumberPagination,
+    _positive_int
+)
 from rest_framework.reverse import reverse_lazy
+from rest_framework.utils.urls import remove_query_param, replace_query_param
+
+
+class DataPagination(PageNumberPagination):
+    """
+    Pagination class for submissions.
+    Because DataViewSet doesn't provide a real Django ORM QuerySet and pagination
+    is based on `start` and `limit` querystring parameters, some methods of
+    PageNumberPagination had to be rewritten to support that case.
+    """
+    page_size_query_param = 'limit'
+    start_query_param = 'start'
+    max_page_size = 500
```

Comment:
Yes and no :-D. 
When pagination is activated, maximum is `max_page_size`. Otherwise it's value on settings. 
But it makes non-sense. I have changed it.